base/set.jl

```diff
@@ -191,8 +191,7 @@ function unique!(f::Callable, A)
             done = true
         end
     end
-    if cur != index  splice!(A, index:cur-1); end
-    A
+    ifelse(cur != index,  resize!(A, index-1), A)
```

Comment:
This calls `resize!` unconditionally.

Reply:
Thanks for the catch.

Actually, thinking more about it... that was a rookie julia user mistake on my part in the use of ifelse... however, interestingly, it does not impact the correctness of the function. 

if 
```resize!(array, length(array))```

is efficient and does not attempt to resize (which seems to be close to the case... when I attempt a resize on a 2_000_000 element array of {Any,1}, it completed in   0.000006 seconds (5 allocations: 176 bytes)

Then, the actual code could just unconditionally return a

```resize!(A, index-1)```

instead of testing.

Reply:
True. `resize!` does that check for you anyway, I'm pretty sure.